### PR TITLE
Libimagstore/walk

### PR DESCRIPTION
**Pull request by matthiasbeyer**
@TheNeikos could you please review this?

### CODE REVIEW

File: libimagstore/src/store.rs [hunk 1]
```diff
@@ -52,6 +54,60 @@ struct StoreEntry {
...
+        self.dirwalker
+            .next()
+            .and_then(|res| {
+                let res = res.map(|next| {
+                    if next.file_type().is_dir() {
+                        Some(StoreObject::Collection(next.path().to_path_buf()))
+                    } else if next.file_type().is_file() {
+                        Some(StoreObject::Id(next.path().to_path_buf()))
+                    } else {
+                        None // TODO: FIXME: Shall we really stop iterating in this case?
```
Comment (TheNeikos) [inline libimagstore/src/store.rs]: Put it in a  loop :)

Comment (matthiasbeyer) [inline libimagstore/src/store.rs]: What do you mean exactly? Do you mean we get the `next()` item in this case and ignore this one?

Comment (TheNeikos) [inline libimagstore/src/store.rs]: Yup? Or am I misunderstanding something?

Comment (matthiasbeyer) [inline libimagstore/src/store.rs]: Nope, could be a solution though I'm not sure I'm happy with it.

**Comment by matthiasbeyer** [comment]
@TheNeikos Could you reiterate for another review? I fixed what we had in the diff discussion and changed the code to loop as we discussed.

**Pull request merged**